store/root.js

```diff
@@ -149,67 +131,13 @@ class Root extends Store {
 
     this.dataProvider = this.user.getDataProviderById(dataProviderInt)
 
-    this.reset()
-    this.retrieveStatistics()
-    this.retrievePluginConfig()
-    this.retrieveIrusStats()
-    this.retrieveRioxxStats()
-
-    const url = `/data-providers/${this.dataProvider.id}`
-    this.works = new Works(url, this.options)
-    this.depositDates = new DepositDates(url, this.options)
-    this.doi = new DOI(url, this.options)
-    this.issues = new Issues(url, this.options)
-  }
-
-  @action
-  async retrieveStatistics() {
-    const url = `/data-providers/${this.dataProvider.id}/statistics`
-    const { data } = await this.request(url)
-    Object.assign(this.statistics, data)
-  }
-
-  @action
-  async retrievePluginConfig() {
-    const url = `/data-providers/${this.dataProvider.id}/plugins`
-    const { data } = await this.request(url)
-
-    data.forEach((plugin) => {
-      this.plugins[plugin.type] = plugin
-    })
-  }
-
-  @action
-  async retrieveIrusStats() {
-    try {
-      const url = `/data-providers/${this.dataProvider.id}/irus`
-      const { data } = await apiRequest(url)
-      this.irus = data
-    } catch (networkOrAccessError) {
-      // Ignore errors for this moment
-    }
-  }
-
-  @action async retrieveRioxxStats() {
-    try {
-      const url = `/data-providers/${this.dataProvider.id}/rioxx/aggregation`
-      const { data } = await apiRequest(url)
-      const {
-        compliantRecordBasic: partiallyCompliantCount,
-        compliantRecordFull: compliantCount,
-        totalRecords: totalCount,
-        ...rest
-      } = data
-
-      this.rioxx = {
-        partiallyCompliantCount,
-        compliantCount,
-        totalCount,
-        ...rest,
-      }
-    } catch (networkOrAccessError) {
-      // Ignore errors for this moment
-    }
+    this.dataProviderData = new DataProvider(
```

Comment:
If you want to have a setter (what you do), you should have a getter too.

It's better to have it under `this.#dataProvider` with `get dataProvider()`. The current name looks confusing. The same applies to the templates.

---

However, I don't think it should be allowed to set the `dataProvider` to a string/number and get an object. If you want to do it right, create private property, a getter and a method like `changeDataProvider` that does all the checks and accepts ID. Otherwise, just use `dataProvider` which can be set to a DataProvider instance by a convention.